### PR TITLE
#166187863 Enable user to follow each other

### PR DESCRIPTION
**Pull request by neelxie**
#### What does this PR do?

Enable authenticated users to follow each other

#### Description of Task to be completed?

- Enable user to view their followers
- Enable one user to follow another author
- Enable user to unfollow author

#### How should this be manually tested?

- git fetch this repo locally in the terminal with $ git fetch origin ft-user-follow-166187863

- install the requirements using pip install -r requirements.txt

- Run the required migrations with `$ python manage.py migrate`

- Run the application server with `$ python manage.py runserver`

- Run this url `http://localhost:8000/swagger/`

- On this page scroll own the the profiles endpoints and try them out.

- Click **Try it out** and enter the ```username``` of the person you want to follow
or unfollow

#### Any background context you want to provide?

For anyone to test out this feature and the endpoints they need to sign up and login atleast two users into the application and then create atleast one article.

#### What are the relevant pivotal tracker stories?

# [166187863](https://www.pivotaltracker.com/story/show/#166187863)

### CODE REVIEW

**Review by HabibSentongo — CHANGES_REQUESTED** [review]
Hey @neelxie, I have gone through your code and it seems fine, but you need to continue working on the coverage and ensure it passes.
The major **problem** I've seen is that we can't access the swagger documentation of your review app so we can't test functionality with it, kindly look into this.
Screenshot 2019-06-19 at 19 59 12

**Comment by Emmanuelaaron** [comment]
Great work @neelxie, on the end point of getting my followers and followees, I do not need to input my username to get these people. Look into this and make it better

**Comment by neelxie** [comment]
> Hey @neelxie, I have gone through your code and it seems fine, but you need to continue working on the coverage and ensure it passes.
> The major **problem** I've seen is that we can't access the swagger documentation of your review app so we can't test functionality with it, kindly look into this.
> Screenshot 2019-06-19 at 19 59 12

I have fixed this error.

**Comment by neelxie** [comment]
> Great work @neelxie, on the end point of getting my followers and followees, I do not need to input my username to get these people. Look into this and make it better

Working on it now.

**Comment by neelxie** [comment]
> Hey @neelxie, I have gone through your code and it seems fine, but you need to continue working on the coverage and ensure it passes.
> The major **problem** I've seen is that we can't access the swagger documentation of your review app so we can't test functionality with it, kindly look into this.
> Screenshot 2019-06-19 at 19 59 12

@HabibSentongo Worked on that and made it better.

**Review by alexxsanya — CHANGES_REQUESTED** [review]
@neelxie kindly update your PR notices with how to test this please. thanks.

**Comment by neelxie** [comment]
> @neelxie kindly update your PR notices with how to test this please. thanks.

@alexxsanya I have.

**Review by alexxsanya — APPROVED** [review]

**Review by alexxsanya — COMMENTED** [review]
@neelxie  I have tested and it's working fine. **Please** work on the feedback which @cdvx gave you of adding a **message** to your response body and we should be good to go.

Screenshot 2019-06-21 at 09 21 18

**Review by Emmanuelaaron — APPROVED** [review]
Great work @neelxie, having rebased from this branch
This is good to go.

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -180,12 +175,12 @@ def post(self, request, **kwargs):
             status=status.HTTP_200_OK,
         )
 
-    @swagger_auto_schema(
-        operation_description='Unfavorite an Article.',
-        operation_id='unfavorite an article',
-        request_body=None,
-        responses={200: serializer_class(many=False), 400: 'BAD REQUEST'},
-    )
+    # @swagger_auto_schema(
```
Comment (cdvx) [inline authors/apps/articles/views.py]: Good work @neelxie , please remove this commented out code

Comment (neelxie) [reply authors/apps/articles/views.py]: I have removed the commented code and re-pushed.

**Pull request merged**